product_customer_code_picking/models/stock_move.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2017 Agile Business Group sagl
+#     (<http://www.agilebg.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.multi
+    def _get_product_customer_code(self):
```

Comment:
@nikul-serpentcs Please follow OCA convention on compute name methods (_compute_...).

Moreover, it is mandatory (even if not stored) to decorate the function with depends.
(picking_id, picking_id.partner_id, product_id.supplier_ids, product_tmpl_id.seller_ids.product_code).

Having this done, you can avoid the search in the loop (which is not performant) and get directly the right supplierinfo through move.product_tmpl_id.seller_ids